langlib/lang.decimal/src/main/java/org/ballerinalang/langlib/decimal/Round.java

```diff
@@ -35,8 +36,16 @@
 //        isPublic = true
 //)
 public class Round {
-
-    public static BDecimal round(BDecimal x) {
-        return ValueCreator.createDecimalValue(x.value().setScale(0, RoundingMode.HALF_EVEN));
+    public static BDecimal round(BDecimal x, long fractionDigits) {
+        BigDecimal value = x.value();
+        BigDecimal scaledDecimal;
+        if (fractionDigits >= value.scale()) {
```

Comment:
Don't we need to give an error for negative `fractionDigits` values?

Reply:
@KavinduZoysa can we check for float case also?

Reply:
https://github.com/ballerina-platform/ballerina-spec/issues/1074